ps2-decompile/callbacks.c

```diff
@@ -1,194 +1,220 @@
                 kb[8 * g->port - 1] = 63;
             }
         }
-        // v4 - 32744 = 2CD4
-        printf((const char *)(v4 + 0xFFFF8018));
     }
+
     return data_transfer(g);
 }
 
 int get_report_done(int result, int count, shared_t *g) {
-    int gp0;              // $gp
-    int transferPipeRet;  // $v0
-    int i;                // [sp+20h] [+20h]
-    int j;                // [sp+20h] [+20h]
-    UsbDeviceRequest req; // [sp+28h] [+28h] BYREF
+    int transferPipeRet;
+    (void) count;
 
+#ifdef DEBUG
     if (result)
-        printf("usbmouse%d: %s -> 0x%x\n", g->port1, "dump_report", result);
+        (void) printf("usbmouse%d: %s -> 0x%x\n", g->port1, "dump_report", result);
+#else
+    (void) result;
+#endif
+
+#ifdef DEBUG
+    int i;
+    // TODO Figure out which values are being compared to detect a descriptor check failure.
     for (i = 0; i < count; ++i) {
         if (*(unsigned char *)(i + 11316) != *(unsigned char *)(i + 0x2D30)) {
-            printf("report desc error\n");
+            (void) printf("report desc error\n");
             g->descriptorChecksPassed = 0;
             break;
         }
     }
-    printf("--Report Descriptor(%d)--\n", count);
+#endif
+
+#ifdef DEBUG
+    int gp0, j;
+    // TODO figure out what its attempting to print out.
+    (void) printf("--Report Descriptor(%d)--\n", count);
     for (j = 0; j < count; ++j) {
-        printf((const char *)(gp0 + 0xFFFF802C), *(unsigned char *)(j + 0x2D30));
+        (void) printf((const char *)(gp0 + 0xFFFF802C), *(unsigned char *)(j + 0x2D30));
         if (j % 2 == 1)
-            printf((const char *)(gp0 + 0xFFFF8034));
+            (void) printf((const char *)(gp0 + 0xFFFF8034));
         if (j % 16 == 15)
-            printf((const char *)(gp0 + 0xFFFF8020));
+            (void) printf((const char *)(gp0 + 0xFFFF8020));
+    }
+#endif
+
+    // sceUsbdControlTransfer(g->configEndpoint, 0x80, USB_REQ_GET_DESCRIPTOR, (USB_DT_STRING << 8) | index, 0x800, ..., g->buf, get_string_done, g)
+    if ((transferPipeRet = sceUsbdTransferPipe(g->configEndpoint, g->buf, 0x800, &REQ_GET_REPORT, get_string_done, g))) {
+#ifdef DEBUG
+        (void) printf("usbmouse%d: sceUsbdSetInterface -> 0x%x\n", g->port1, transferPipeRet)
+#endif
     }
-    printf((const char *)(gp0 + 0xFFFF8020));
-    req.requesttype = 0x80; // USB_DIR_IN
-    req.request = 6;        // USB_REQ_GET_DESCRIPTOR
-    req.value = 768;
-    req.index = 0;
-    req.length = 0x800;
-    transferPipeRet = sceUsbdTransferPipe(
-        g->configEndpoint,
-        g->buf,
-        0x800,
-        &req,
-        get_string_done,
-        g); // sceUsbdControlTransfer(g->configEndpoint, 0x80, USB_REQ_GET_DESCRIPTOR, (USB_DT_STRING << 8) | index, 0x800, ..., g->buf, get_string_done, g)
-    if (transferPipeRet)
-        transferPipeRet =
-            printf("usbmouse%d: %s -> 0x%x\n", g->port1, "sceUsbdSetInterface", transferPipeRet);
+
```

Comment:
Why are the above lines deleted?

Reply:
line 80 it resides; the request was predefined in the header since its just static values.